File: programmers/01.hash/03.전화번호 목록/python/01.py
#  효율성 실패 
def solution(phone_book):
    phone_book.sort(key=len)
    for i, num1 in enumerate(phone_book):
        for num2 in (phone_book[i+1:]):
            if num1 == num2[:len(num1)]:  
                return False
        
    return True



# test
print(solution([ "97674223", "1195524421","119"]))
print(solution(["123","456","789"]))
print(solution(["12","123","1235","567","88"]))

File: programmers/01.알고리즘/04.sort/03.H-index/01.js
// h이상인 수가 h개

function solution(citations) {
    citations.sort((a, b) => b - a);
    // console.log(citations);

    let cnt = 0;
    while (cnt + 1 <= citations[cnt]) {
        cnt++;
    }
    return cnt;
}

// test
console.log(solution([3, 0, 6, 1, 5]));
